ethcore/src/client/client.rs

```diff
@@ -276,6 +278,25 @@ impl Client {
 		self.notify.write().push(Arc::downgrade(&target));
 	}
 
+	/// Set a closure to call when we want to restart the client
+	pub fn set_exit_handler<F>(&self, f: F) where F: Fn(bool, Option<String>) + 'static + Send {
+		*self.exit_handler.lock() = Some(Box::new(f));
+	}
+
+	/// Hypervisor should restart this client.
+	pub fn restart(&self) {
+		if let Some(ref h) = *self.exit_handler.lock() {
+			(*h)(true, None);
+		}
+	}
+
+	/// Hypervisor should kill this client.
+	pub fn exit(&self) {
```

Comment:
`exit` and `restart` are not yet called from anywhere, however, they seem like useful public api to have; i plan on implementing a UI endpoints for them to all the user to kill or restart parity easily.